### ISSUE TITLE
Format selector `best[ext=webm]/bestvideo*[ext=webm]+bestaudio*[ext=webm]/best` is not working as expected

### ISSUE THREAD

**Issue by benoit74** [yt-dlp/yt-dlp#11189]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

I'm trying to download a Youtube video thanks to `yt_dlp` Python package.

For convenience, I've reproduced the issue with CLI.

Following command randomly fails: `yt_dlp --format "best[ext=webm]/bestvideo*[ext=webm]+bestaudio*[ext=webm]/best" https://www.youtube.com/watch\?v\=7_N0yozUnWY`.

Error message is `Requested format is not available`.

From my tests of `yt_dlp --list-formats https://www.youtube.com/watch\?v\=7_N0yozUnWY`, it looks like depending on which Youtube server we hit, there is sometimes no webm video at all, but there is always many `mp4` formats available.

I would have expected the format selector to fallback to best `mp4` since it finishes with `/best`. I would obviously prefer `webm` but I prefer `mp4` than nothing. 

Do I miss something or is there a bug?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
> yt_dlp -vU --format "best[ext=webm]/bestvideo*[ext=webm]+bestaudio*[ext=webm]/best" https://www.youtube.com/watch\?v\=7_N0yozUnWY
[debug] Command-line config: ['-vU', '--format', 'best[ext=webm]/bestvideo*[ext=webm]+bestaudio*[ext=webm]/best', 'https://www.youtube.com/watch?v=7_N0yozUnWY']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.10.01.001408 from yt-dlp/yt-dlp-master-builds [e59c82a74] (darwin_exe)
[debug] Python 3.12.6 (CPython arm64 64bit) - macOS-14.5-arm64-arm-64bit (OpenSSL 3.0.15 3 Sep 2024)
[debug] exe versions: ffmpeg 7.1 (setts), ffprobe 7.1
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, sqlite3-3.45.3, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest
Latest version: master@2024.10.01.001408 from yt-dlp/yt-dlp-master-builds
yt-dlp is up to date (master@2024.10.01.001408 from yt-dlp/yt-dlp-master-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=7_N0yozUnWY
[youtube] 7_N0yozUnWY: Downloading webpage
[youtube] 7_N0yozUnWY: Downloading ios player API JSON
[youtube] 7_N0yozUnWY: Downloading web creator player API JSON
[youtube] 7_N0yozUnWY: Downloading m3u8 information
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
ERROR: [youtube] 7_N0yozUnWY: Requested format is not available. Use --list-formats for a list of available formats
Traceback (most recent call last):
  File "yt_dlp/YoutubeDL.py", line 1626, in wrapper
  File "yt_dlp/YoutubeDL.py", line 1782, in __extract_info
  File "yt_dlp/YoutubeDL.py", line 1841, in process_ie_result
  File "yt_dlp/YoutubeDL.py", line 2977, in process_video_result
yt_dlp.utils.ExtractorError: [youtube] 7_N0yozUnWY: Requested format is not available. Use --list-formats for a list of available formats
```

**Comment by seproDev** [comment]
The `web_creator` client yt-dlp currently still uses as a default, is currently getting a change rolled out that requires sign in. We will likely change the default soon.
In the cases where the `web_creator` is unavailable, only the `ios` client is used, which yields these formats:
```
ID  EXT   RESOLUTION FPS CH │   FILESIZE   TBR PROTO │ VCODEC          VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sb2 mhtml 48x27        0    │                  mhtml │ images                                  storyboard
sb1 mhtml 80x45        0    │                  mhtml │ images                                  storyboard
sb0 mhtml 160x90       0    │                  mhtml │ images                                  storyboard
233 mp4   audio only        │                  m3u8  │ audio only          unknown             [fr] Default
234 mp4   audio only        │                  m3u8  │ audio only          unknown             [fr] Default
139 m4a   audio only      2 │    4.63MiB   49k https │ audio only          mp4a.40.5   49k 22k [fr] low, m4a_dash
140 m4a   audio only      2 │   12.28MiB  129k https │ audio only          mp4a.40.2  129k 44k [fr] medium, m4a_dash
251 webm  audio only      2 │    9.84MiB  104k https │ audio only          opus       104k 48k [fr] medium, webm_dash
269 mp4   256x144     25    │ ~ 11.27MiB  119k m3u8  │ avc1.4D400C    119k video only
160 mp4   256x144     25    │    3.56MiB   38k https │ avc1.4D400C     38k video only          144p, mp4_dash
230 mp4   640x360     25    │ ~ 38.99MiB  411k m3u8  │ avc1.4D401E    411k video only
134 mp4   640x360     25    │   14.17MiB  150k https │ avc1.4D401E    150k video only          360p, mp4_dash
605 mp4   640x360     25    │ ~ 28.83MiB  304k m3u8  │ vp09.00.21.08  304k video only
232 mp4   1280x720    25    │ ~109.26MiB 1153k m3u8  │ avc1.64001F   1153k video only
136 mp4   1280x720    25    │   39.45MiB  416k https │ avc1.64001F    416k video only          720p, mp4_dash
270 mp4   1920x1080   25    │ ~172.02MiB 1815k m3u8  │ avc1.640028   1815k video only
137 mp4   1920x1080   25    │   69.28MiB  731k https │ avc1.640028    731k video only          1080p, mp4_dash
```
`best[ext=webm]` - Fails since there exists no premerged video with webm extension
`bestvideo*[ext=webm]+bestaudio*[ext=webm]` - Fails since there exists no video only format with webm extension
`best` - Fails since there exists no premerged format.

Format selection is working correctly. `best` means premerged format. Most streaming sites don't offer those anymore.

**Comment by benoit74** [comment]
Crystal clear, thank you very much !

**Comment by bashonly** [comment]
Fixed by de2062753a188060d76f587e45becce61fe399f9
